app/src/main/java/dev/animeshvarma/sigil/ui/screens/CustomEncryptionScreen.kt

```
@@ -54,12 +62,38 @@ import dev.animeshvarma.sigil.ui.theme.AnimationConfig
+ * The UI exposes controls for toggling compression, reordering/removing encryption layers, adding layers via a bottom sheet,
+ * editing input and password (with vault integration), triggering encryption/decryption, and viewing/sharing/copying output.
+ * It also presents a Save Profile dialog (including optional KDF override and raw mode) and an overwrite confirmation dialog
+ * when a profile name already exists.
+ *
+ * @param viewModel The SigilViewModel providing state, actions, and persistence APIs used by the screen.
+ * @param uiState Current UiState containing layers, input/password/output text, compression flag, and editing profile id.
+ */
 @Suppress("AssignedValueIsNeverRead")
 @OptIn(ExperimentalMaterial3Api::class, ExperimentalFoundationApi::class)
 @Composable
 fun CustomEncryptionScreen(viewModel: SigilViewModel, uiState: UiState) {
```

detekt (1.23.8)

[warning] The function CustomEncryptionScreen appears to be too complex based on Cognitive Complexity (complexity: 52). Defined complexity threshold for methods is set to '25'

(detekt.complexity.CognitiveComplexMethod)